app/models/shipping_rate.rb

```diff
@@ -27,6 +27,14 @@ def usps_rates
 
   private
 
+  def format_ups_rates(rates)
+    rates_hash = {}
+    rates.sort_by(&:price).map do |rate|
+      rates_hash[rate.service_name] = {"price" => rate.total_price, "delivery_date" => rate.delivery_date}
```

Comment:
I'm also confused in terms of what's happening here -- format_ups_rates should be returning an array so that an order can be maintained.